webapp/sources/ldap-inventory/inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/PreInventoryParserCheckInventoryAge.scala

```diff
@@ -0,0 +1,129 @@
+   * - ACCESSLOG > LOGDATE, which is a datetime in local server time, without timezone.
+   * - OPERATINGSYSTEM > TIMEZONE > TIMEZONE that may not be there, for more fun.
+   *
+   * If we are not able to parse the timezone, then assume UTC because it's at least a known error.
+   * If logdate is missing or not parsable, assume the inventory is out of range (it will likely
+   * be refused further one).
+   */
+  override def apply(inventory: NodeSeq): IOResult[NodeSeq] = {
+    val now  = OffsetDateTime.now()
+    val date = PreInventoryParserCheckInventoryAge.extracDateValue(inventory)
+
+    (for {
+      d <- PreInventoryParserCheckInventoryAge.parseInventoryDate(date)
+      _ <- PreInventoryParserCheckInventoryAge.checkDate(d, now, maxBeforeNow, maxAfterNow)
+    } yield inventory).toIO
+  }
+
+}
+
+object PreInventoryParserCheckInventoryAge {
+
+  val dateTimeFormat       = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+  val offsetDateTimeFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssZ")
+
+  def extracDateValue(inventory: NodeSeq): String = {
+    val logdate  = optText(inventory \\ "ACCESSLOG" \ "LOGDATE").getOrElse("")
+    val timezone = optText(inventory \\ "OPERATINGSYSTEM" \ "TIMEZONE" \ "OFFSET").getOrElse("")
+
+    logdate + timezone
+  }
+
+  // try to parse the inventory date in format YYYY-MM-dd HH:mm:ss+oooo
+  // ex UTC: 2023-11-16 10:46:35+0000
+  // ex Europe/Paris: 2023-11-16 11:46:12+0100
+  def parseOffsetDateTime(date: String): PureResult[OffsetDateTime] = {
+    PureResult.attempt(s"Error when parsing date '${date}' as a time-zoned date with offset")(
+      OffsetDateTime.parse(date, offsetDateTimeFormat)
+    )
+  }
+
+  // try to parse the inventory date in format: YYYY-MM-dd HH:mm:ss
+  // Since in that case, we don't have timezone information, we need to chose arbitrary,
+  // and we chose UTC so that it's easier to identify problem when they arise.
+  def parseLocalDateTime(date: String): PureResult[OffsetDateTime] = {
+    PureResult
+      .attempt(s"Error when parsing date '${date}' as a local date time")(
+        LocalDateTime.parse(date, dateTimeFormat)
+      )
+      .map(OffsetDateTime.of(_, ZoneOffset.UTC))
+  }
+
+  def parseInventoryDate(date: String): PureResult[OffsetDateTime] = {
+    parseOffsetDateTime(date) match {
+      case Left(_)  => parseLocalDateTime(date)
+      case Right(d) => Right(d)
+    }
+  }
+
+  // check that date is in wundow between [now-maxPast, now+maxFuture]
+  def checkDate(date: OffsetDateTime, now: OffsetDateTime, maxBeforeNow: Duration, maxAfterNow: Duration): PureResult[Unit] = {
+    val oldest   = now.minus(maxBeforeNow)
+    val youngest = now.plus(maxAfterNow)
+
+    val tooOld    = Inconsistency(s"Inventory is too old, refusing (inventory date is before '${oldest.toString}')")
+    val tooFuture = Inconsistency(s"Inventory is too far in the future, refusing (inventory date is before '${oldest.toString}')")
```

Comment:
For the `tooFuture`, should it not be the `youngest` in the error message to know the maximum acceptation date ?

Reply:
totally, bad replace after copy/paste